pkg/operator/controllers/machineset/machineset_controller_test.go

```diff
@@ -0,0 +1,415 @@
+				if err != nil {
+					t.Error(err)
+				}
+
+				if *modifiedMachineset.Spec.Replicas != tt.wantReplicas {
+					t.Error(cmp.Diff(*modifiedMachineset.Spec.Replicas, tt.wantReplicas))
+				}
+			}
+		})
+	}
+}
+
+func TestCustomConfiguration(t *testing.T) {
+	// Test the reconciler when there is a custom machineset present
+	newFakeMao := func(replicas, customReplicas int32) *maofake.Clientset {
+		workerMachineSet := &machinev1beta1.MachineSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "aro-fake-machineset-0",
+				Namespace: machineSetsNamespace,
+				Labels: map[string]string{
+					"machine.openshift.io/cluster-api-machine-role": "worker",
+				},
+			},
+			Spec: machinev1beta1.MachineSetSpec{
+				Replicas: to.Int32Ptr(replicas), // Modify replicas accordingly
+			},
+		}
+		customMachineSet := &machinev1beta1.MachineSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "custom-machineset",
+				Namespace: machineSetsNamespace,
+				Labels: map[string]string{
+					"machine.openshift.io/cluster-api-machine-role": "worker",
+				},
+			},
+			Spec: machinev1beta1.MachineSetSpec{
+				Replicas: to.Int32Ptr(customReplicas),
+			},
+		}
+		return maofake.NewSimpleClientset(workerMachineSet, customMachineSet)
+	}
+
+	baseCluster := arov1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{Name: arov1alpha1.SingletonClusterName},
+		Spec: arov1alpha1.ClusterSpec{
+			InfraID:  "aro-fake",
+			Features: arov1alpha1.FeaturesSpec{ReconcileMachineSet: true},
+		},
+	}
+
+	tests := []struct {
+		name         string
+		maocli       *maofake.Clientset
+		wantReplicas int32
+		wantErr      string
+		objectName   string
+	}{
+		// run the same tests, but expect no changes or errors since we have introduced a custom machineset
+		{
+			name:         "no worker replicas, machineset-0 modified",
+			maocli:       newFakeMao(0, 0),
+			wantReplicas: 0,
+			wantErr:      "",
+			objectName:   "aro-fake-machineset-0",
+		},
+		{
+			name:         "one worker replica, machineset-0 modified",
+			maocli:       newFakeMao(1, 0),
+			wantReplicas: 1,
+			wantErr:      "",
+			objectName:   "aro-fake-machineset-0",
+		},
+		{
+			name:         "two worker replicas, machineset-0 modified",
+			maocli:       newFakeMao(1, 1),
+			wantReplicas: 1,
+			wantErr:      "",
+			objectName:   "aro-fake-machineset-0",
+		},
+		{
+			name:         "three worker replicas, machineset-0 modified",
+			maocli:       newFakeMao(3, 0),
+			wantReplicas: 3,
+			wantErr:      "",
+			objectName:   "aro-fake-machineset-0",
+		},
+		{
+			name:         "three worker replicas in machineset-1, machineset-0 modified",
+			maocli:       newFakeMao(0, 3),
+			wantReplicas: 0,
+			wantErr:      "",
+			objectName:   "aro-fake-machineset-0",
+		},
+		{
+			name:         "two worker replicas in machineset-1, custom-machineset modified",
+			maocli:       newFakeMao(0, 2),
+			wantReplicas: 2,
+			wantErr:      "",
+			objectName:   "custom-machineset",
+		},
```

Comment:
I think all the test cases can be replaced with a single test case in `TestReconciler`: in that case we need to have less than 2 replicas and a custom machine set and make sure that if we modify - nothing changes.